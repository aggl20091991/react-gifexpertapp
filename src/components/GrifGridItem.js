import React from 'react'
import PropTypes from 'prop-types';

function GrifGridItem({id,title,url}) {
  return (
    <div className="card animate__animated animate__bounce">
        <img src={url} alt={title}/>
        <p>{title}</p>
    
    </div>
  )
}
GrifGridItem.prototypes = {
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
}

export default GrifGridItem