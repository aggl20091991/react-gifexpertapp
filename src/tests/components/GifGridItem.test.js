import React from 'react'
import { shallow } from "enzyme";
import GrifGridItem from "../../components/GrifGridItem";

describe('pruebas en gifgriditem/>', () => {
    const title ='cualquier titulo'
    const url = 'https://google.com'
    const wrapper = shallow(<GrifGridItem title={title} url={url} />)

    test('debe mostrar el componente correctaente', () => { 
        expect(wrapper).toMatchSnapshot();

     })

     test('debe tener un parrafo', () => { 
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)         

      })
      test('should url same', () => { 
        const img = wrapper.find('img')
         // console.log(img.html())
          expect(img.prop('src')).toBe(url)
          expect(img.prop('alt')).toBe(title)
          
       })
       test('should has animate', () => { 
        const div = wrapper.find('div')
         // console.log(div.prop())
          const className = div.prop('className')
          expect(className.includes('animate__bounce')).toBe(true)
           
        })
})