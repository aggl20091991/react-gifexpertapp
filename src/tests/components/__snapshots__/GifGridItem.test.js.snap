// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en gifgriditem/> debe mostrar el componente correctaente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="cualquier titulo"
    src="https://google.com"
  />
  <p>
    cualquier titulo
  </p>
</div>
`;
